src/components/instance-form-builder/GeneralInfoForm.jsx

```diff
@@ -166,12 +166,21 @@ class GeneralInfoForm extends React.Component {
                 return;
             }
 
-            this.setState({ isSaving: true });
+            this.props.instance.validateInstanceName(this.props.d2).then(isValid => {
+                if (!isValid.status) {
```

Comment:
Here we have a unique validation and can be kept as it is. As a rule of thumb, it's preferable to have a single  `model.validate()` that returns something like: `[[FIELD, [msg1, msg2]], [FIELD2, [msg3, msg4]]`. This way the component has less logic.

Reply:
done!